test/integration/src/test/java/io/pravega/test/integration/CheckpointTest.java

```diff
@@ -951,4 +952,122 @@ public void testGenerateStreamCuts() throws Exception {
+        ReaderGroupConfig groupConfigSC = ReaderGroupConfig.builder()
+                                                           .disableAutomaticCheckpoints()
+                                                           .stream(Stream.of(scope, streamName), StreamCut.UNBOUNDED, streamCut)
+                                                           .build();
+        streamManager.createScope(scope);
+        streamManager.createStream(scope, streamName, StreamConfiguration.builder()
+                                                                         .scalingPolicy(ScalingPolicy.fixed(2))
+                                                                         .build());
+
+        streamManager.createReaderGroup(readerGroupName, groupConfigSC);
+        @Cleanup
+        ReaderGroup readerGroup = streamManager.getReaderGroup(readerGroupName);
+        JavaSerializer<String> serializer = new JavaSerializer<>();
+        @Cleanup
+        EventStreamWriter<String> producer = clientFactory.createEventWriter(streamName, serializer,
+                EventWriterConfig.builder().build());
+        producer.writeEvent("seg0", testString);
+        producer.writeEvent("seg0", testString1);
+        producer.writeEvent("seg0", testString2);
+        producer.writeEvent("seg0", testString3);
+        producer.writeEvent("seg1", testString);
+        producer.writeEvent("seg1", testString1);
+        producer.writeEvent("seg1", testString2);
+        producer.writeEvent("seg1", testString3);
+
+        producer.flush();
+
+        AtomicLong clock = new AtomicLong();
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(readerName, readerGroupName, serializer,
+                ReaderConfig.builder().build(), clock::get,
+                clock::get);
+        EventStreamReader<String> reader1 = clientFactory.createReader("reader1", readerGroupName, serializer,
+                ReaderConfig.builder().build(), clock::get,
+                clock::get);
+
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader.readNextEvent(1000).getEvent());
+        assertNotNull(reader1.readNextEvent(1000).getEvent());
+        assertNotNull(reader1.readNextEvent(1000).getEvent());
+        assertNotNull(reader1.readNextEvent(1000).getEvent());
+        assertNotNull(reader1.readNextEvent(1000).getEvent());
+        //Initiating 1st checkpoint
+        clock.addAndGet(CLOCK_ADVANCE_INTERVAL);
+        final InlineExecutor backgroundExecutor = new InlineExecutor();
+        CompletableFuture<Checkpoint> cpreader1 = readerGroup.initiateCheckpoint("CheckpointReader1", backgroundExecutor);
+        assertFalse(cpreader1.isDone());
+        // reader read next event at checkpoint
+        EventRead<String> read = reader.readNextEvent(500);
```

Comment:
Here and below, there is not much difference between 500ms and 100 or 200 because both are more than fast enough for network latency but not so fast so as to iterate twice through the loop. Anywhere a value is NOT expected, it's worth while to use a low timeout to speed up the test execution.

Reply:
Changed timeout to 100ms for null event check